wordcount/wordcount.go

```diff
@@ -1,62 +1,125 @@
 package main
 
 import (
+	"log"
 	"github.com/pauloaguiar/ces27-lab1/mapreduce"
 	"hash/fnv"
+	"regexp"
+	"strings"
+	"strconv"
 )
 
-// mapFunc is called for each array of bytes read from the splitted files. For wordcount
-// it should convert it into an array and parses it into an array of KeyValue that have
-// all the words in the input.
+/**
+ * mapFunc is called for each array of bytes read from the splitted files. For wordcount
+ * it should convert it into an array and parses it into an array of KeyValue that have
+ * all the words in the input.
+ */
 func mapFunc(input []byte) (result []mapreduce.KeyValue) {
-	// 	Pay attention! We are getting an array of bytes. Cast it to string.
-	//
-	// 	To decide if a character is a delimiter of a word, use the following check:
-	//		!unicode.IsLetter(c) && !unicode.IsNumber(c)
-	//
-	//	Map should also make words lower cased:
-	//		strings.ToLower(string)
-	//
-	// IMPORTANT! The cast 'string(5)' won't return the character '5'.
-	// 		If you want to convert to and from string types, use the package 'strconv':
-	// 			strconv.Itoa(5) // = "5"
-	//			strconv.Atoi("5") // = 5
+	/*
+	 * Pay attention! We are getting an array of bytes. Cast it to string.
+	 *
+	 * To decide if a character is a delimiter of a word, use the following check:
+	 * !unicode.IsLetter(c) && !unicode.IsNumber(c)
+	 *
+	 * Map should also make words lower cased:
+	 * 	strings.ToLower(string)
+	 *
+	 * IMPORTANT! The cast 'string(5)' won't return the character '5'.
+	 *		If you want to convert to and from string types, use the package 'strconv':
+	 *			strconv.Itoa(5) // = "5"
+	 *			strconv.Atoi("5") // = 5
+	 */
 
-	/////////////////////////
-	// YOUR CODE GOES HERE //
-	/////////////////////////
+	/*
+	 * Casting byte array to string
+	 */
+	s := string(input[:])
```

Comment:
The [:] is not necessary here, as it's return a new slice that is exactly the same as input.